src/api/ZoweExplorerApi.ts

```diff
@@ -42,7 +42,7 @@ export namespace ZoweExplorerApi {
          *      will use the profile the API was retrieved with by default
          * @returns {Session} a Zowe CLI Session
          */
-        getSession(profile?: IProfileLoaded): Session;
+        getSession(profile?: IProfileLoaded, prompt?: boolean): Promise<Session>;
```

Comment:
The API is designed to be independent of a specific UI implementation. Add this and the zOSMF implementation below introduces a cyclic dependency to the UI.  The UI code should be extracted into it own class and made independent of z/OSMF anyway using the meta_data files for determining what to prompt for.